pkg/apis/experimental/v1alpha1/custom_builder_validation.go

```diff
@@ -24,11 +26,37 @@ func (cb *CustomBuilder) Validate(ctx context.Context) *apis.FieldError {
 
 func (s *CustomBuilderSpec) Validate(ctx context.Context) *apis.FieldError {
 	return validate.Tag(s.Tag).
-		Also(validate.FieldNotEmpty(s.Stack, "stack")).
-		Also(validate.FieldNotEmpty(s.Store, "store"))
+		Also(validateStack(s.Stack).ViaField("stack")).
+		Also(validateStore(s.Store).ViaField("store"))
 }
 
 func (s *CustomNamespacedBuilderSpec) Validate(ctx context.Context) *apis.FieldError {
 	return s.CustomBuilderSpec.Validate(ctx).
 		Also(validate.FieldNotEmpty(s.ServiceAccount, "serviceAccount"))
 }
+
+func validateStack(stack v1.ObjectReference) *apis.FieldError {
+	if stack.Name == "" {
+		return apis.ErrMissingField("name")
+	}
+
+	switch stack.Kind {
+	case ClusterStackKind:
+		return nil
+	default:
+		return apis.ErrInvalidValue(stack.Kind, "kind")
```

Comment:
Should we default this to "ClusterStack"?

Reply:
I think yes!